src/analyz/mod.rs

```diff
@@ -790,18 +790,41 @@ fn init_instances_from_tests(ms: &mut MirState, out: &mut impl JsonOutput) -> io
             continue;
         }
 
-        if !([DefKind::Fn, DefKind::AnonConst].contains(&tcx.def_kind(def_id))) {
-            tcx.sess.span_err(
-                tcx.def_span(def_id),
-                "#[test] can only be applied to functions",
-            );
-            continue;
+        // Unless the DefId corresponds to a function, throw an error. We make
+        // an effort to cast a wide net in catching things that look like
+        // functions, as saw-rustc will try to initialize _all_ functions, not
+        // just the ones that users opt into, as is the case with crux-rustc.
+        let non_function_err = || {
+        };
```

Comment:
```suggestion
```
This closure appears to be unused.

Reply:
Sigh, this is a holdover from a previous version of the patch that I failed to clean up afterwards. Fixed.